qutip/core/cy/coefficient.pyx

```diff
@@ -387,14 +387,23 @@ cdef class InterCoefficient(Coefficient):
     order : int
         Order of the interpolation. Order ``0`` uses the previous (i.e. left)
         value. The order will be reduced to ``len(tlist) - 1`` if it is larger.
+
+    bc_type : 2-Tuple or None, optional
+        Boundary conditions for spline evaluation. Default value is `None` to
+        enable automatically choosing boundary conditions. Otherwise, a two-length
+        tuple of form (`deriv_l`, `deriv_r`) must be passed. Here, `deriv_l` denotes
+        the boundary conditions at ```t = 0``` and deriv_r, the conditions at time
+        ```t```. The condition specified must be an iterable pair of form (`order`,
+        `value`). Refer to Scipy's documentation for further details:
+        https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.make_interp_spline.html
```

Comment:
I would suggest to shorten this and leave it to scipy's doc:
```
boundary_conditions : 2-Tuple, str or None, optional
    Boundary conditions for spline evaluation. Default value is `None`. 
    Correspond to `bc_type` of scipy.interpolate.make_interp_spline.
    Refer to Scipy's documentation for further details:
    https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.make_interp_spline.html
```

Otherwise, there is an error in the description: the times seems to be forced to be between `t=0` and `t`. But `tlist` can start and stop at any values as long as they are increasing. Also the explanation for the str values are missing.